WordPress/Classes/Utility/Media/ImageDownloader.swift

```diff
@@ -53,13 +53,21 @@ class ImageDownloader {
     @discardableResult
     func downloadImage(for request: URLRequest, completion: @escaping (UIImage?, Error?) -> Void) -> ImageDownloaderTask {
         let task = session.dataTask(with: request) { (data, _, error) in
-            guard let data = data, let image = UIImage(data: data) else {
-                let error = error ?? ImageDownloaderError.failed
```

Comment:
Aren't we loosing the chance to pass a better error?

Reply:
Good catch!